File: javascript/allergies/_iterations/1/allergies.js
const ALLERGIES = {
    1: 'eggs',
    2: 'peanuts',
    4: 'shellfish',
    8: 'strawberries',
    16: 'tomatoes',
    32: 'chocolate',
    64: 'pollen',
    128: 'cats'
};

export class Allergies {
    constructor(score) {
        this.allergies = Object.keys(ALLERGIES).filter(is => (score & Number(is)) !== 0)
                                               .map(is => ALLERGIES[is]);
    }

    list() {
        return this.allergies;
    }

    allergicTo(allergen) {
        return this.allergies.indexOf(allergen) != -1;
    }
}

File: clojure/allergies/_iterations/1/src/allergies.clj
(ns allergies)

(def ^:private allergens
  [:eggs :peanuts :shellfish :strawberries :tomatoes :chocolate :pollen :cats])

(defn- allergic-to-score? [score allergen-score]
  (pos? (bit-and score (bit-shift-left 1 allergen-score))))

(defn allergies [score]
  (->> (range (count allergens))
       (filter (partial allergic-to-score? score))
       (map (partial get allergens))))

(defn allergic-to? [score allergen]
  (allergic-to-score? score (.indexOf allergens allergen)))
